analyses/independent-samples/independent-samples.R

```diff
@@ -44,6 +44,11 @@ independent_samples <- function(sample_df,
       dplyr::filter(Kids_First_Participant_ID %in% no_primary)
     sample_df <- dplyr::bind_rows(primary_df, noprimary_df)
   } 
+  
+  if(tumor_types == "secondary"){
+    sample_df <- sample_df %>%
+      dplyr::filter(Kids_First_Participant_ID %in% no_primary)
+  } 
```

Comment:
The `no_primary` variable has list of samples which only have secondary and no primary samples. 
For example the Kids_First_Participant_ID == `PT_YGN06RPZ` has both primary (BS_3C4YAMKJ) and secondary (BS_060QB7XB) samples and if we filter by `no_primary` we will miss out on the capturing the secondary sample. 

Our requirement here is a list of samples with all secondary samples, so I think we would find the secondary sample list by filtering samples by tumor_descriptors not in `primary_descs`. 
```suggestion
if(tumor_types == "secondary"){
    sample_df <- sample_df %>%
      dplyr::filter(!tumor_descriptor %in% primary_descs)
  } 
```

Reply:
Thanks @kgaonkar6 for clarification! I will update the code and output files accordingly.